cylc/flow/data_store_mgr.py

```diff
@@ -1524,7 +1524,8 @@ def update_data_structure(self, reloaded=False):
             # Gather this batch of deltas for publish
             self.publish_deltas = self.get_publish_deltas()
 
-        self.updates_pending = next_update_pending
+        self.updates_pending = self.next_update_pending
```

Comment:
If `self.next_update_pending` is only used to set `self.updates_pending`, can we merge these variables.

Reply:
The problem is; the `update_data_structure` does things in a specific order (which we can change):
- Prune data-store (remove from internal sets and create pruned deltas for application to the store)
- Update family proxies (states and state totals updated (including root), **based on current DB**
- Update the workflow fields (including state and state totals, by collating all the states of the cycle roots)
- Prune deltas of pruned nodes (so they don't get applied)
- Batch all deltas
- Apply deltas to data-store
- publish deltas

So the issue is that the state_totals get applied before the prune deltas get applied to the data-store (before the old data gets removed, hence why it's only succeeded counts).. 
So what I was trying to do with `self.next_update_pending` is run the update to redo the state-totals (..etc) after a pruning has been applied.

So a couple of options would be:
- Apply the pruning to the data-store first then do the state totals (and just make sure we manage the deltas properly)
- Ignore those-to-be-pruned in the state-totals calculations

The latter is proving to be a little tricky.. so I might try the former ..

Reply:
Actually strike that .. I've found the issue